lib/plurimath/math/formula.rb

```diff
@@ -996,6 +996,12 @@ def upcase_dd_intent_name(nodes)
         encode(str)
       end
       # Dd derivative nodes end
+
+      def type_error!
+        raise Math::InvalidTypeError.new(
+          "`type` must be one of: `#{MATH_ZONE_TYPES.join('`, `')}`",
```

Comment:
Can the message also write out that value was given?

Reply:
@ronaldtse, I have updated the message, please let me know if you have suggestions.